src/common/transformations/src/transformations/utils/utils.cpp

```diff
@@ -491,11 +492,20 @@ bool is_constant_and_all_values_equal_int(const Output<Node>& output, const int6
 
 bool is_on_constant_path(const ov::Output<ov::Node>& output) {
     auto status = true;
-    std::deque<ov::Node*> nodes_to_calculate = {output.get_node()};
+
+    auto root_node = output.get_node();
+    if (!root_node || root_node->get_output_size() == 0) {
+        return false;
+    }
+    std::deque<ov::Node*> nodes_to_calculate = {root_node};
 
     while (status && !nodes_to_calculate.empty()) {
         auto current_node = nodes_to_calculate.front();
         nodes_to_calculate.pop_front();
+        // RandomUniform output changes during runtime, so we should not consider it as a constant
+        if (current_node->get_type_info() == ov::op::v8::RandomUniform::get_type_info_static()) {
```

Comment:
do we have any other operations we know that might break Constant calculation similar way as RandomUniform? 
or is it not clear right now?

Reply:
as we synced offline, I think currently we only observed this 'RandomUniform' Op break Constant calculation

Reply:
if we face similar cases, we will extend the api of this function with a new "stop (break) nodes" argument